v2/cmd/controller/main.go

```diff
@@ -170,7 +173,18 @@ func makeControllerOptions(log logr.Logger, cfg config.Values) controllers.Optio
 				// TODO: do we need GVK here too?
 				return log.WithValues("namespace", req.Namespace, "name", req.Name)
 			},
+			// These rate limits are used for happy-path backoffs (for example polling async operation IDs for PUT/DELETE)
 			RateLimiter: controllers.NewRateLimiter(1*time.Second, 1*time.Minute),
 		},
+		RequeueIntervalCalculator: interval.NewCalculator(
+			// These rate limits are primarily for ReadyConditionImpactingError's
+			interval.CalculatorParameters{
+				//nolint:gosec // do not want cryptographic randomness here
+				Rand:         rand.New(lockedrand.NewSource(time.Now().UnixNano())),
+				BaseDelay:    1 * time.Second,
+				MaxFastDelay: 30 * time.Second,
+				MaxSlowDelay: 3 * time.Minute,
```

Comment:
Is 3m slow enough? We _could_ make this 5m but that's pretty darn long

Reply:
The Slow delay is used for Azure stuff, right? Some Azure things - like provisioning and deletion - can take a super long time.

That said, 3m = 20 requests per hour, and 5m = 12 requests per hour, so the difference isn't that great in the general scheme of things.

Reply:
Will improve the comments here to clearly indicate. Really `MaxFast` and `MaxSlow` are for errors only. So when we're getting an error from Azure, we'll back off to one request every 3m on retries. That'd be the case for example for 5xx's or 4xx level errors that we're retrying on.

The happy path always takes the upper ratelimiter time (I hate that there are 2 but there have to be to make this work unfortunately). So when things are working correctly and we're polling an async operation, we'll poll every 1m. AFAIK, polling async operations even by the SDKs is done at a pretty snappy pace, so backing off to 3m there seems unnecessary. The compute API for example specifically says something along the lines of: "if you want to poll fast, poll the async operation ID, don't issue GETs")

Reply:
Changed these names to `ErrorMaxFastDelay` and `ErrorMaxSlowDelay` to clarify where they are actually used.

Reply:
Will leave this at 3 minutes for now.